Mathlib/Data/Matrix/Basic.lean

```diff
@@ -1889,6 +1889,27 @@ theorem vecMul_one (v : m → α) : v ᵥ* 1 = v := by
   rw [← diagonal_one, vecMul_diagonal, mul_one]
 #align matrix.vec_mul_one Matrix.vecMul_one
 
+@[simp]
+theorem diagonal_mulVec (x : α) (v : m → α) : (diagonal fun _ => x) *ᵥ v = x • v := by
```

Comment:
I guess maybe this should be `diagonal_const_mulVec` to avoid confusion with `mulVec_diagonal`.

Reply:
```suggestion
theorem scalar_mulVec [Semiring α] (x : α) (v : m → α) : (Matrix.scalar m α) *ᵥ v = x • v := by
```